examples/stable-diffusion/text_to_image_generation.py

```diff
@@ -320,6 +325,14 @@ def main():
     if args.throughput_warmup_steps is not None:
         kwargs_call["throughput_warmup_steps"] = args.throughput_warmup_steps
 
+    if args.use_cpu_rng:
+        # Patch for the deterministic generation - Need to specify CPU as the torch generator
+        generator = [
+            torch.Generator(device="cpu").manual_seed(args.seed + i) for i in range(args.num_images_per_prompt)
```

Comment:
This will fail when you have more than 1 prompt.   You would need to use `args.num_images_per_prompt * len(args.prompts)` instead.   But I am trying to understand why not just use `generator = torch.Generator(device="cpu").manual_seed(args.seed)` here?  Are we trying to demonstrate an example that you can set generator for each sample separately?

Reply:
Synced with @dsocek . It will be better to just pass one CPU generator instead of a list. I have modified the code.